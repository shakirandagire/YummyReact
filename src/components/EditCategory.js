import React, { Component } from 'react';
import axios from 'axios';
import { notify } from 'react-notify-toast';
import { BASE_URL } from '../../src/constants';
import Navigation from './Navigation';

/**
 * Component for editing categories
 */
class EditCategory extends Component {
    state = {
      categoryname: '',
      category_description: '',
    }
    // Component for populating the fields of the edit form
    componentWillMount() {
      const headers = { Authorization: `Bearer ${localStorage.getItem('accessToken')}` };
      const categoryId = this.props.match.params.id;
      axios
        .get(`${BASE_URL}/api/v1/categories/${categoryId}`, { headers })
        .then((response) => {
          const category = response.data;
          this.setState({
            categoryname: category.categoryname,
            category_description: category.category_description });
        })
        .catch((error) => {
          if (error.response) {
            notify.show(error.response.data.message, 'success');
          } else if (error.request) {
            notify.show("Request not made");
          }
        });
    }

      handleInputChange = (event) => {
        const { name, value } = event.target;
        this.setState({ [name]: value });
      }

      // Function to handle the editing of the categories

      handleEditCategory = (event) => {
        const headers = { Authorization: `Bearer ${localStorage.getItem('accessToken')}` };
        event.preventDefault();
        const categoryId = this.props.match.params.id;
        axios
          .put(`http://127.0.0.1:5000/api/v1/categories/${categoryId}`, this.state, { headers })
          .then(() => {
            notify.show("Category edited successfully", 'success', 4000);
            this.props.history.push('/view-categories');
          })
          .catch((error) => {
            if (error.response) {
              notify.show(error.response.data.message, 'success');
            } else if (error.request) {
              notify.show("Request not made");
            }
          });
      }

      render() {
        return (
          <div className="my_container backgimg">
            <Navigation />

            <div className="mycontent">

              <form onSubmit={this.handleEditCategory}>

                <div className="form-group">
                  Category Name:
                  <input type="text" name="categoryname" className="form-control" required onChange={this.handleInputChange} value={this.state.categoryname} />
                </div>
                <div className="form-group">
                  Description:
                  <input type="text" name="category_description" className="form-control" required onChange={this.handleInputChange} value={this.state.category_description} />
                </div>

                <button type="submit" className="btn btn-danger formsave" >Save</button>
              </form>
            </div>

          </div>


        );
      }
}
export default EditCategory;
