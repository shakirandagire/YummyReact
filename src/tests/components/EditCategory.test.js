import React from 'react';
import EditCategory from '../../components/editcategories';
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';



describe('<EditCategory/>',() => {
    const params = {
        match: {
            params: {
              id: 1,
            },
          },
        };
  const component = shallow(< EditCategory match={{params}}/>);
  it('should render properly', () =>{
    expect(component.length).toBe(1)
  })

  it('should render edit categories',() =>{
    expect(shallowToJson(component)).toMatchSnapshot();
})

})

