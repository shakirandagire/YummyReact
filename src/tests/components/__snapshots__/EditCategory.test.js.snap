// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<EditCategory/> should render edit categories 1`] = `
<div
  className="my_container backgimg"
>
  <nav
    className="navbar navbar-inverse"
  >
    <div
      className="container-fluid"
    >
      <div
        className="navbar-header"
      >
        <a
          className="navbar-brand title"
          href="#"
        >
          Yummy Recipe App
        </a>
      </div>
      <ul
        className="nav navbar-nav title"
      >
        <li>
          <a>
            Home
          </a>
        </li>
        <li>
          <a>
            Recipes
          </a>
        </li>
        <li>
          <a>
            Categories
          </a>
        </li>
      </ul>
      <ul
        className="nav navbar-nav navbar-right title"
      >
        <li>
          <a>
            <span
              className="glyphicon glyphicon-user"
            />
             Hallo
          </a>
        </li>
        <li>
          <a>
            Logout
          </a>
        </li>
      </ul>
    </div>
  </nav>
  <div
    className="mycontent"
  >
    <form
      onSubmit={[Function]}
    >
      <div
        className="form-group"
      >
        <label>
          Category Name:
        </label>
        <input
          className="form-control"
          name="categoryname"
          onChange={[Function]}
          required={true}
          type="text"
          value=""
        />
      </div>
      <div
        className="form-group"
      >
        <label>
          Description:
        </label>
        <input
          className="form-control"
          name="category_description"
          onChange={[Function]}
          required={true}
          type="text"
          value=""
        />
      </div>
      <button
        className="btn btn-danger formsave"
        type="submit"
      >
        Save
      </button>
    </form>
  </div>
</div>
`;
